src/Datasource/Paging/NumericPaginator.php

```diff
@@ -389,6 +390,9 @@ protected function buildParams(array $data): array
         $this->pagingParams['limit'] = $data['defaults']['limit'] != $data['options']['limit']
             ? $data['options']['limit']
             : null;
+        if (isset($data['options']['maxLimit'])) {
```

Comment:
Since `maxLimit` is initialized to null I don't think this isset() check is necessary. We can always set `maxLimit` for paging params.